### DISCUSSION TITLE
Format for RGB contract media kit API

### DISCUSSION THREAD

**Discussion by dr-orlovsky** [LNP-BP/LNPBPs#162] [Proposals]
RGB contracts can be shipped together with a supplement (a part of the consignment) signed by a developer. This can be a contract developer - or wallet developer, or any other party whose signature a user trusts.

As a part of the supplement, we have a "media kit" - a URL pointing to where a wallet and other software may retrieve actual icons, images, colours and other things related to the contract.

I propose to define a response of the server on that URL as an LNPBP standard. I propose that the server should respond with JSON object with the following structure:
```json
{
    "dark": {
      "icon": {
          "32x32": URL,
          "64x64": URL,
      },
      "banner": {
          "468x256": URL,
          // ...
      },
      "transparent": { /*...*/ } // transparent background image
      "bgColor": "#ad7834c" // in CSS format,
      "fgColor": ...
    }
}
```

Any other ideas?

**Comment by cryptoquick** [comment]
Why can't this information be encoded into the supplement itself?

**Reply by dr-orlovsky** [reply]
Since it can change (more sizes and formats can be added over time, more themes). Yes, issuer or wallet can distribute new supplements, but it will be more complex task than just using web here.

**Reply by fedsten** [reply]
Also to me it seems a bit an overkill to try to standardize such minor stuff. I would just leave the freedom to issuer to encode these kind of information into the supplement itself as they wish to (if they desire to do so at all).

**Reply by dr-orlovsky** [reply]
@fedsten and how wallets would understand different encodings from different issuers and use them?
